node/service/src/lib.rs

```diff
@@ -1361,3 +1369,63 @@ pub fn build_full(
 	#[cfg(not(feature = "polkadot-native"))]
 	Err(Error::NoRuntime)
 }
+
+struct RevertConsensus {
+	blocks: BlockNumber,
+	backend: Arc<FullBackend>,
+}
+
+impl ExecuteWithClient for RevertConsensus {
+	type Output = sp_blockchain::Result<()>;
+
+	fn execute_with_client<Client, Api, Backend>(self, client: Arc<Client>) -> Self::Output
+	where
+		<Api as sp_api::ApiExt<Block>>::StateBackend: sp_api::StateBackend<BlakeTwo256>,
+		Backend: sc_client_api::Backend<Block> + 'static,
+		Backend::State: sp_api::StateBackend<BlakeTwo256>,
+		Api: polkadot_client::RuntimeApiCollection<StateBackend = Backend::State>,
+		Client: AbstractClient<Block, Backend, Api = Api> + 'static,
+	{
+		babe::revert(client.clone(), self.backend, self.blocks)?;
+		grandpa::revert(client, self.blocks)?;
+		Ok(())
+	}
+}
+
+/// Reverts the node state down to at most the last finalized block.
+#[cfg(feature = "full-node")]
+pub fn revert(
```

Comment:
nit: we already have bunch of `fn revert`, a better name - or just a descriptive suffix - would be helpful.

Reply:
Damn names :-) Does `revert_backend` or `revert_backend_aux` fits?
I'm open to suggestions ;-)